### PR TITLE
docs/flow: add monitoring docs

### PR DESCRIPTION
**Pull request by rfratto**
This PR adds documentation for how to monitor and debug Grafana Agent Flow. 

The UI is described here, but I've purposefully left screenshots out for the time being until the UI is merged in main (since some details may change in the meantime).

### CODE REVIEW

File: docs/flow/monitoring/debugging.md
Comment (rfratto) [inline docs/flow/monitoring/debugging.md:15]: It makes sense for us to put screenshots somewhere in this section, but I'd like to hold off on doing that until the UI is merged in main just in case details change during review.

File: docs/flow/monitoring/component_metrics.md
Comment (karengermond) [inline docs/flow/monitoring/component_metrics.md:19]: Should this just be "Component-specific metrics are exposed..."?  Are they actually metrics for metrics?

File: docs/flow/monitoring/debugging.md
Comment (karengermond) [inline docs/flow/monitoring/debugging.md:12]:
```suggestion
* Through the Grafana Agent Flow UI
```

File: docs/flow/monitoring/debugging.md
Comment (karengermond) [inline docs/flow/monitoring/debugging.md:13]:
```suggestion
* Through examining logs
```

File: docs/flow/monitoring/debugging.md
Comment (karengermond) [reply docs/flow/monitoring/debugging.md:12]: Suggest reserving numbered lists for steps that need a specific order, otherwise use bullets.

File: docs/flow/monitoring/debugging.md
Comment (karengermond) [inline docs/flow/monitoring/debugging.md:25]:
```suggestion
* The home page shows a table of running components along with their health.
```

File: docs/flow/monitoring/debugging.md
Comment (karengermond) [inline docs/flow/monitoring/debugging.md:28]:
```suggestion
* The "Graph" page shows a graph view of all running components along with
```

File: docs/flow/monitoring/debugging.md
Comment (karengermond) [inline docs/flow/monitoring/debugging.md:26]:
```suggestion
   Clicking the "View" button navigates to the detail page for that
```

File: docs/flow/monitoring/debugging.md
Comment (karengermond) [inline docs/flow/monitoring/debugging.md:29]:
```suggestion
   their health. Clicking a component in the graph navigates to the
```

File: docs/flow/monitoring/debugging.md
Comment (karengermond) [inline docs/flow/monitoring/debugging.md:32]:
```suggestion
The component detail page shows the following information for each
```

File: docs/flow/monitoring/debugging.md
Comment (karengermond) [inline docs/flow/monitoring/debugging.md:35]:
```suggestion
* The health of the component with a message explaining the health.
```

File: docs/flow/monitoring/debugging.md
Comment (karengermond) [inline docs/flow/monitoring/debugging.md:36]:
```suggestion
* The current evaluated arguments for the component.
```

File: docs/flow/monitoring/debugging.md
Comment (karengermond) [inline docs/flow/monitoring/debugging.md:37]:
```suggestion
* The current exports for the component.
```

File: docs/flow/monitoring/debugging.md
Comment (karengermond) [inline docs/flow/monitoring/debugging.md:38]:
```suggestion
* The current debug info for the component (if the component has debug info).
```

File: docs/flow/monitoring/debugging.md
Comment (karengermond) [inline docs/flow/monitoring/debugging.md:52]:
```suggestion
To reduce logging noise, many components hide debugging info behind
```

File: docs/flow/monitoring/debugging.md
Comment (karengermond) [inline docs/flow/monitoring/debugging.md:53]:
```suggestion
debug-level log lines. It is recommended that you configure the [`logging
```

File: docs/flow/monitoring/debugging.md
Comment (karengermond) [inline docs/flow/monitoring/debugging.md:54]: Is this supposed to be emit or omit?  Just making sure.

File: docs/flow/monitoring/debugging.md
Comment (karengermond) [inline docs/flow/monitoring/debugging.md:58]:
```suggestion
Refer to the [`logging block`][logging] page to see how to find logs for your
```

**Review by karengermond — APPROVED** [review]
LGTM

File: docs/flow/monitoring/debugging.md
Comment (rfratto) [reply docs/flow/monitoring/debugging.md:12]: Right, the intention here is that people check the UI first before trying to investigate logs.

File: docs/flow/monitoring/debugging.md
Comment (rfratto) [reply docs/flow/monitoring/debugging.md:54]: Yep, emit is right here: we recommend enabling debug logs so users see everything while debugging. 

Should we pick another word here?

File: docs/flow/monitoring/component_metrics.md
Comment (rfratto) [reply docs/flow/monitoring/component_metrics.md]: > Should this just be "Component-specific metrics are exposed..."?

Oops, yes :)

File: docs/flow/monitoring/debugging.md
Comment (rfratto) [reply docs/flow/monitoring/debugging.md:12]: Should that be more clear in the preceding sentence?

File: docs/flow/monitoring/debugging.md
Comment (rfratto) [reply docs/flow/monitoring/debugging.md]: I've changed this to show, LMK if you think that sounds ok :)

**Review by tpaschalis — APPROVED** [review]
LGTM, not much to add here!

File: docs/flow/monitoring/debugging.md
Comment (tpaschalis) [inline docs/flow/monitoring/debugging.md:13]: Do we want to also add a third bullet-point, where users might want to check the component-level metrics?

File: docs/flow/monitoring/debugging.md
Comment (mattdurham) [inline docs/flow/monitoring/debugging.md:31]: Do we need to note that any secret values will be obfuscated in the UI?

Comment (rfratto) [reply docs/flow/monitoring/debugging.md:31]: Good idea, I'll throw that in a blockquote somewhere as an aside 👍

**Review by mattdurham — APPROVED** [review]
Minor nits but LGTM

File: docs/flow/monitoring/debugging.md
Comment (rfratto) [reply docs/flow/monitoring/debugging.md:13]: Good idea, I'll write a section for that 👍

Comment (rfratto) [reply docs/flow/monitoring/debugging.md:13]: Actually, thinking about it, I don't know if component-level metrics helps _debug_ issues (rather, they're more useful for _identifying_ issues). WDYT?

File: docs/flow/monitoring/debugging.md
Comment (rfratto) [inline docs/flow/monitoring/debugging.md:29]: UI elements should be referenced in bold rather than quotation marks: https://developers.google.com/style/ui-elements#formatting

(cc @karengermond for a heads up, I stumbled on this while reading the style guide)

File: docs/flow/monitoring/debugging.md
Comment (rfratto) [reply docs/flow/monitoring/debugging.md:13]: For now, I'm going to leave out checking metrics since I'm not positive that's ever helpful for resolving/debugging issues

**Pull request merged**
